tests/manage/pv_services/test_resource_deletion_during_pvc_pod_creation_and_io.py

```diff
@@ -0,0 +1,286 @@
+            log.info(
+                f"Read: {fio_result.get('jobs')[0].get('read').get('iops')}"
+            )
+            log.info(
+                f"Write: {fio_result.get('jobs')[0].get('write').get('iops')}"
+            )
+        log.info("Verified IO result on pods.")
+
+        # Delete pods
+        all_pod_objs = io_pods + pod_objs_new
+        for pod_obj in all_pod_objs:
+            pod_obj.delete(wait=False)
+
+        # Verify pods are deleted
+        for pod_obj in all_pod_objs:
+            pod_obj.ocp.wait_for_delete(resource_name=pod_obj.name)
+
+        # Updating self.pod_objs for the purpose of teardown
+        self.pod_objs.clear()
+
+        # Verify that PVCs are reusable by creating new pods
+        all_pvc_objs = self.pvc_objs + pvc_objs_new
+        pod_objs_re = helpers.create_pods(
+            pvc_objs_list=all_pvc_objs, interface_type=self.interface,
+            desired_status=constants.STATUS_RUNNING, wait=True,
+            namespace=self.namespace
+        )
+        log.info("Successfully created new pods using all PVCs.")
+
+        # Updating self.pod_objs for the purpose of teardown
+        self.pod_objs.extend(pod_objs_re)
+
+        # Run IO on each of the newly created pods
+        for pod_obj in pod_objs_re:
+            pod_obj.run_io(
+                storage_type='fs', size='100M', runtime=10,
+                fio_filename='fio-file-retest'
+            )
+
+        log.info("Fetching IO results from newly created pods")
+        for pod_obj in pod_objs_re:
+            get_fio_rw_iops(pod_obj)
+        log.info("Verified IO result on newly created pods.")
+
+
+@tier4
+@pytest.mark.usefixtures(
+    create_rbd_secret.__name__,
+    create_ceph_block_pool.__name__,
+    create_rbd_storageclass.__name__,
+    create_pvcs.__name__,
+    test_fixture.__name__
+)
+class TestForRBD(OperationsBase):
+    """
+    Test class for RBD
+    """
+    interface = constants.CEPHBLOCKPOOL
+
+    @pytest.mark.parametrize(
+        argnames="resource_to_delete",
+        argvalues=[
+            pytest.param(
+                *['mgr'],
+                marks=pytest.mark.polarion_id("OCS-735")
+            ),
+            pytest.param(
+                *['mon'],
+                marks=pytest.mark.polarion_id("OCS-736")
+            ),
+            pytest.param(
+                *['osd'],
+                marks=pytest.mark.polarion_id("OCS-737")
+            )
+
+        ]
+    )
+    def test_resource_deletion_during_pvc_pod_creation_and_io_block(
+        self, resource_to_delete
+    ):
+        """
+        Delete resource while PVC creation, Pods creation, IO are progressing.
+        RBD PVC
+        """
+        self.operations_base(resource_to_delete)
+
+
+@tier4
+@pytest.mark.usefixtures(
+    create_cephfs_secret.__name__,
+    create_cephfs_storageclass.__name__,
+    create_pvcs.__name__,
+    test_fixture.__name__
+)
+class TestForCephFS(OperationsBase):
```

Comment:
Can we give meaningful name for Test Class

Reply:
Done